script/hassfest/coverage.py

```diff
@@ -31,11 +95,34 @@ def validate(integrations: Dict[str, Integration], config: Config):
             path = Path(line)
 
             # Discard wildcard
-            while "*" in path.name:
-                path = path.parent
+            path_exists = path
+            while "*" in path_exists.name:
+                path_exists = path_exists.parent
 
-            if not path.exists():
+            if not path_exists.exists():
                 not_found.append(line)
+                continue
+
+            if (
+                not line.startswith("homeassistant/components/")
+                or not len(path.parts) == 4
+                or not path.parts[-1] == "*"
+            ):
+                continue
+
+            integration_path = path.parent
+
+            integration = integrations[integration_path.name]
+
+            for check in DONT_IGNORE:
+                if (integration_path.name, check) in ALLOWED_IGNORE_VIOLATIONS:
+                    continue
+
+                if (integration_path / check).exists():
+                    integration.add_error(
+                        "coverage",
+                        f"coveragerc incorrectly set up to ignore coverage for {check}",
```

Comment:
I think the current message tells not what is wrong.

```
Invalid integrations: 1

Found errors. Generating files canceled.

Integration acer_projector:
* [COVERAGE] coveragerc incorrectly set up to ignore coverage for config_flow.py
```

Maybe this should say:

```
* [COVERAGE] config_flow.py should not be ignored by the coveragerc file
```

Maybe we should even state we require this file to have tests? As in general, those are missing at that point as well.

Reply:
Good point. 

They are not actually always missing!

Reply:
Sorry, I should have been more clear.

We can check if the tests are missing, by the filename. For example, if `test_config_flow.py` is missing, that could trigger a different message here.

It is quite often we have to point to that. As coverage failure isn't always clear to people what is entails.